app/models/manageiq/providers/openstack/event_catcher_mixin.rb

```diff
@@ -1,14 +1,7 @@
 module ManageIQ::Providers::Openstack::EventCatcherMixin
   extend ActiveSupport::Concern
 
-  class_methods do
-    def all_valid_ems_in_zone
-      require 'manageiq/providers/openstack/legacy/openstack_event_monitor'
-      super.select do |ems|
-        ems.sync_event_monitor_available?.tap do |available|
-          _log.info("Event Monitor unavailable for #{ems.name}.  Check log history for more details.") unless available
-        end
-      end
-    end
+  def self.all_valid_ems_in_zone
+    super.select { |ems| ems.supports?(:events) }
```

Comment:
Tempted to move this check to the core EventCatcher class and add `supports :events` to all providers which have it.  Then only providers which need to run a check (e.g. openstack/azure) need an instance block like https://github.com/ManageIQ/manageiq-providers-openstack/pull/805/files#diff-6e82d48dbcba5ccd5c9dee5a5f722c52485f2c738505fe1698d54d9786e69fa0R55